Add index library and README (#8)
Nrevyw/wyvern-public-resources: index-library/README.md

# Index Library
This directory contains files relevant to Wyvern's Hyperspectral Index Library. Currently, the extent of this library

> is a single `json` file containing all of our indices. This may grow in the future

Reviewer: ```suggestion
is a single `json` file containing all of our indices. This may grow in the future.
```

> is a single `json` file containing all of our indices. This may grow in the future

Reviewer 2: ```suggestion
is a single `json` file containing all of our indices. This may grow in the future.
```

`wyvern_index_library.json`, with some additional information within the PR:
- Relevant bands. Please use [our product guide](https://guide.wyvern.space/data-product-guide/product-specifications/spectral-bands)
for more information on our bands
- Relevant papers and resources to include with the index. (We will only include indices that have relevant
scientific papers attached to them)

### JSON Schema
```json
{
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "$id": "https://example.com/product.schema.json",
    "$defs": {
        "hsiBand": {
            "description": "Represents a single HSI band including band parameters (FWHM, etc). Used to link back to equations.",
            "type": "object",
            "properties": {
                "band_index": {
                    "description": "Index of band in geotiff/file. Starts from 1",
                    "type": "integer"
                },
                "cwl_nm": {
                    "description": "Centre wavelength of band in nanometres",
                    "type": "integer"
                },
                "fwhm_nm": {

>                     "description": "Full-width-half-max (FWHM) of band. I.e. bandwidth.",

Reviewer: For consistency w/line 50.

```suggestion
                    "description": "Full-width-half-max (FWHM) of band (i.e. bandwidth).",
```

                    "type": "number"
                },
                "approximate_match": {
                    "description": "If true, it only matches the required band in the equation approximately (i.e. over 5nm away from band).",
                    "type": "boolean"
                }
            }
        }
    },
    "title": "Wyvern Index Library Item",
    "description": "An item in the Wyvern hyperspectral index library",
    "type": "object",
    "properties": {
        "name": {
            "description": "Name of the hyperspectral index",
            "type": "string"
        },
        "description": {
            "description": "Rich text description of the hyperspectral index, including relevant caveats/notes. Add as much colour as possible here!",
            "type": "string"
        },
        "categories": {

>             "description": "Array of categories. These should be high level like: vegetation, water, geology, etc.",

Reviewer: If you're using `i.e.`, get the two-fer and throw in `e.g.` as well. 🤣 

```suggestion
            "description": "Array of categories. These should be high level (e.g. vegetation, water, geology, etc).",
```

            "type": "array",
            "items": {
                "type": "string"
            },
            "minItems": 1,
            "uniqueItems": true
        },
        "keywords": {

>             "description": "Array of keywords. These should be focused like: chlorophyll, stress, iron, broadband, normalized difference, etc",

Reviewer: ```suggestion
            "description": "Array of keywords. These should be focused (e.g.  chlorophyll, stress, iron, broadband, normalized difference, etc).",
```

            "type": "array",
            "items": {
                "type": "string"
            },
            "minItems": 1,
            "uniqueItems": true
        },
        "equation": {
            "description": "Simple text equation for the HSI index",
            "type": "string"
        },
        "latex_equation": {
            "description": "LaTEX/KaTEX compatible equation for rich rendering",
            "type": "string"
        },
        "band_mappings": {
            "description": "Band mapping objects for each compatible satellite",
            "type": "object",
            "properties": {
                "Dragonette-001": {
                    "description": "Band mapping for Dragonette-001",
                    "type": "object",
                    "propertyNames": {
                        "pattern": ".*"
                    },
                    "additionalProperties": {
                        "$ref": "#/$defs/hsiBand"
                    }
                },

>                 "Dragonette-2/3/4": {

Reviewer 2: If we made this 'Extended VNIR' with a 'title' of 'Dragonette-2/3/4', it would prevent a breaking change when we add Dragonette-5.

Author: You're right.. let me rip the bandaid off and just do it now.

>                     "description": "Band mapping for Dragonette-002/3/4/+. These satellites have the same band set.",

Reviewer: Suggestion, non-blocking: maybe take out the `+` to avoid confusion, since it's not in the field name.
